### PR TITLE
DELTASPIKE-1176 Support void methods in Futureable

### PR DESCRIPTION
**Pull request by kalgon**

### CODE REVIEW

File: deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/future/DefaultFutureableStrategy.java [hunk 1]
```diff
@@ -202,6 +203,13 @@ public Object call() throws Exception
         };
 
         final ExecutorService pool = getOrCreatePool(ic);
+        
+        if (void.class.isAssignableFrom(returnType))
```
Comment (johnament) [inline deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/future/DefaultFutureableStrategy.java]: Should we also check for `Void` in addition to `void`?

Comment (r-tt) [inline deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/future/DefaultFutureableStrategy.java]: think `void` is enough, using Void is a workaround IMO and not a clean programming model we should promote. If we get some feedbacks justifying this we can add it later.

Alternative is to add a `Handler` depending the returned type and a kind of SPI but looks overkilled _today_.

Comment (rmannibucau) [inline deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/future/DefaultFutureableStrategy.java]: think void is enough, using Void is a workaround IMO and not a clean programming model we should promote. If we get some feedbacks justifying this we can add it later.

Alternative is to add a Handler depending the returned type and a kind of SPI but looks overkilled today.

Comment (johnament) [inline deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/future/DefaultFutureableStrategy.java]: good to merge then?

Comment (rmannibucau) [inline deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/future/DefaultFutureableStrategy.java]: for me yes

**Comment by johnament** [comment]
Changes merged, however I had to rebase.  Please close this PR.

**Comment by seanf** [comment]
This was merged as https://github.com/apache/deltaspike/commit/17b3513f69f3546f4b60f8a5ca744644983944a9 in time for **deltaspike-1.7.2**, for anyone else who goes looking.

This should be very handy, thanks!

**Pull request closed without merging**